front/app/containers/Admin/projects/project/general/components/GeographicAreaInputs.tsx

```diff
@@ -14,11 +14,51 @@ import { IAreaData } from 'services/areas';
 import { useParams } from 'react-router-dom';
 import { TOnProjectAttributesDiffChangeFunction } from '..';
 
+// styles
+import styled from 'styled-components';
+import { fontSizes, colors } from 'utils/styleUtils';
+
 interface Props {
   areaIds: string[] | undefined;
   onProjectAttributesDiffChange: TOnProjectAttributesDiffChangeFunction;
 }
 
+const LabelText = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin-top: -2px;
+
+  &.disabled {
+    opacity: 0.4;
+    cursor: not-allowed;
+  }
+
+  .header {
+    padding: 0;
+    margin: 0;
+    margin-bottom: 3px;
+    font-weight: 600;
+    font-size: ${fontSizes.base}px;
+  }
+
+  .description {
+    color: ${colors.adminSecondaryTextColor};
+  }
```

Comment:
You won't need these styles if you import the component that already exists (see below).

Reply:
`LabelHeaderDescription` is in `front/app/containers/Admin/projects/project/participationContext/components/labels.tsx` which directly imports `front/app/containers/Admin/projects/project/messages.ts` (so it doesn't accept other strings or message files). I can centralize that component but that's more effort when generally I think it's ok to copy/paste things once or twice until it's obvious it needs to be centralized and/or generalized.

I can also do  `import { LabelText } from 'containers/Admin/projects/project/participationContext/components/styling'` but a similar issue, we're either now coupled to the component in participationContext or need to move it to a central place, generalize it, and make sure it works properly in both locations, not just the one in the initial ticket

Reply:
If you just change the original LabelHeaderDescription's props to accept the JSX.element like you have with the duplicated LabelHeaderDescription's, you don't need to worry about the location of messages.ts, right? It can stay exactly where it is. For LabelHeaderDescription, you can create a components folder in /projects/project and also put the LabelText in the same file as your LabelHeaderDescription. No need for it to be in that separate file, it's only used in 1 place.

At least, if you don't generalize the component, you can import it from participationContext, so it's clear for the next person that it's the same thing. You can quite easily change the other place where it's used to have JSX.Element for the props as well. I like how you did it much better, because you have some type checking on messages.

I'm being persistent here because I don't want to accept unmaintainable code anymore. I want to spend my time on building features, not fighting debt, so we need to raise the bar and sweat the small stuff. This solution is going to create more work for the next person to refactor, and maybe they'll pass it on to the next person then as well because it doesn't fit with their small task.

Also, typing these messages takes longer than making the changes so we should just do it next time.

Reply:
Sure, I'll make the requested changes.